apps/core/declarativewebcontainer.cpp

```diff
@@ -1040,7 +1042,7 @@ void DeclarativeWebContainer::updateLoading()
 
 void DeclarativeWebContainer::updateActiveTabRendered()
 {
-    if (m_webPage && !m_webPage->completed() && !m_webPage->domContentLoaded()) {
+    if (m_webPage && !m_webPage->completed() && !m_webPage->domContentLoaded() && !m_webPage->isPainted()) {
```

Comment:
```suggestion
    if (!m_webPage || !m_webPage->completed() || (!m_webPage->isPainted() && !m_webPage->domContentLoaded())) {
```
Either isPainted or domContentLoaded is enough for the update.